Refactor extra properties mapping logic in Mapperly

salihozkara opened the pull request.

salihozkara committed:
Fix EF Core change tracking issue by creating a new dictionary instance during mapping

salihozkara committed:
Add tests

maliming commented:
hi @salihozkara 

All unit tests passed even after I reverted your change. Can you add a unit test to verify your code?

Thanks.

<img width="2328" height="1858" alt="image" src="https://github.com/user-attachments/assets/32312ba1-5455-49d9-a1a0-4f35e064b348" />

salihozkara committed:
Refactor extra properties mapping logic in Mapperly

Simplified the logic for mapping extra properties by removing redundant reference checks and always invoking MapExtraProperties with appropriate defaults. Updated related tests and removed unnecessary [MapExtraProperties] attributes from sample mappers.
